dojo/tools/edgescan/parser.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+from dojo.models import Endpoint, Finding
+from dojo.tools.edgescan.importer import EdgescanImporter
+
+ES_SEVERITIES = {1: "Info", 2: "Low", 3: "Medium", 4: "High", 5: "Critical"}
+SCAN_EDGESCAN_API = 'Edgescan API Scan'
+SCAN_EDGESCAN = 'Edgescan Scan'
+
+
+class EdgescanParser(object):
+    """
+    Import from Edgescan API or JSON file
+    """
+
+    def get_scan_types(self):
+        return [SCAN_EDGESCAN, SCAN_EDGESCAN_API]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type
+
+    def get_description_for_scan_types(self, scan_type):
+        if scan_type == SCAN_EDGESCAN_API:
+            return "Edgescan findings can be imported by API."
+        return "Edgescan findings can be imported in JSON format."
+
+    def requires_file(self, scan_type):
+        return False if scan_type == SCAN_EDGESCAN_API else True
+
+    def requires_tool_type(self, scan_type):
+        return 'Edgescan' if scan_type == SCAN_EDGESCAN_API else None
+
+    def get_findings(self, file, test):
+        try:
+            if file:
+                serialized_data = file.read()
+                try:
+                    data = json.loads(str(serialized_data, "utf-8"))
+                except:
+                    data = json.loads(serialized_data)
+            else:
+                data = EdgescanImporter().get_findings(test)
+        except:
+            raise Exception("Invalid details provided")
+
+        return self.process_vulnerabilities(test, data)
+
+    def process_vulnerabilities(self, test, vulnerabilities):
+        findings = []
+
+        for vulnerability in vulnerabilities:
+            findings.append(self.make_finding(test, vulnerability))
+
+        return findings
+
+    def make_finding(self, test, vulnerability):
+        finding = Finding(test=test)
+        finding.title = vulnerability["name"]
+        finding.date = vulnerability["date_opened"][:10]
+        if vulnerability["cwes"]:
+            finding.cwe = int(vulnerability["cwes"][0][4:])
+        if vulnerability["cves"]:
+            finding.cve = str(vulnerability["cves"])
+        finding.url = vulnerability["location"]
+        finding.severity = ES_SEVERITIES[vulnerability["severity"]]
+        finding.description = vulnerability["description"]
+        finding.mitigation = vulnerability["remediation"]
+        finding.active = True
+        finding.verified = True
+        finding.false_p = False
+        finding.duplicate = False
+        finding.out_of_scope = False
+        finding.numerical_severity = Finding.get_numerical_severity(ES_SEVERITIES[vulnerability["severity"]])
+        finding.tags = vulnerability["asset_tags"]
+        finding.unique_id_from_tool = vulnerability["id"]
+
+        finding.unsaved_endpoints = [Endpoint.from_uri(vulnerability["location"])]
+
+        return finding
```

Comment:
It seems that there are CVSS vectors in the reports, you need to implement `cvssv3 ` with the vector
![image](https://user-images.githubusercontent.com/1694940/148683543-8afe00dc-4063-4b49-ab8b-b3a8c1315ae4.png)

Reply:
Fixed in 387fc77. If the cvss version is v3 then it will extract the vector and score otherwise it is left blank.

Reply:
If the cvss version is v3 then the vector and score are added otherwise, it is left blank.